orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForCapacityMatchTask.groovy

```diff
@@ -40,11 +40,30 @@ class WaitForCapacityMatchTask extends AbstractInstancesCheckTask {
 
   @Override
   protected boolean hasSucceeded(Stage stage, Map serverGroup, List<Map> instances, Collection<String> interestingHealthProviderNames) {
-    if (!serverGroup.capacity || serverGroup.capacity.desired != instances.size()) {
-      return false
+    def splainer = new WaitForUpInstancesTask.Splainer()
+      .add("Capacity match check for server group ${serverGroup?.name} [executionId=${stage.execution.id}, stagedId=${stage.execution.id}]")
+
+    try {
+      if (!serverGroup.capacity) {
+        splainer.add("short-circuiting out of WaitForCapacityMatchTask because of empty capacity in serverGroup=${serverGroup}")
+        return false
+      }
+
+      splainer.add("checking if capacity matches (capacity.desired=${serverGroup.capacity.desired}, instances.size()=${instances.size()}) ")
+      if (serverGroup.capacity.desired != instances.size()) {
+        splainer.add("short-circuiting out of WaitForCapacityMatchTask because expected and current capacity don't match}")
+        return false
+      }
+
+      if (serverGroup.disabled) {
+        splainer.add("capacity matches but server group is disabled, so returning hasSucceeded=true")
+        return true
+      }
+
+      splainer.add("capacity matches and server group is enabled, so we delegate to WaitForUpInstancesTask to check for healthy instances")
+      return WaitForUpInstancesTask.allInstancesMatch(stage, serverGroup, instances, interestingHealthProviderNames, splainer)
+    } finally {
```

Comment:
TIL you don't need to `catch` a `try` block, neat

Reply:
> What happens if something throws an exception before we call `splainer.splain()`?

My reading on how this works is `splainer.splain()` will get called, _then_ the exception will be thrown up the stack, since it is not being caught in this block. So we'll get whatever accumulation of logging messages have been recorded.

Reply:
would it be worth catching an exception here and splainin' that the action failed due to an exception?

Reply:
I believe you're right @anotherchrisberry, `splainer.splain()` is in the `finally` block to make sure we `splain()` even if we return early or throw an exception, and then the exception should bubble up as it would normally.

Good point about logging the exception @cfieber!